src/lib/support/ThreadOperationalDataset.h

```diff
@@ -270,88 +191,219 @@ class OperationalDataset
+     * The required elements are: PAN ID, Extended PAN ID, Channel, and Master Key.
+     */
+    bool IsCommissioned() const;
+
+    /**
+     * This method checks if the dataset is empty.
+     */
+    bool IsEmpty() const { return mData.empty(); }
+
+    /**
+     * This method checks whether @p aData contains a valid sequence of Thread TLV elements.
      *
-     * @retval CHIP_NO_ERROR on success.
-     * @retval CHIP_ERROR_NO_MEMORY if there is insufficient space within the dataset.
+     * @note This method only verifies the overall TLV format, not the correctness of individual TLV elements.
      */
-    CHIP_ERROR SetSecurityPolicy(uint32_t aSecurityPolicy);
+    static bool IsValid(ByteSpan aData);
+
+private:
+    friend class OperationalDataset;
+
+    ByteSpan mData;
+
+    OperationalDatasetView(uint8_t * buffer) : mData(ByteSpan(buffer, 0)) {}
+
+    const ThreadTLV * Locate(uint8_t aType) const;
+    bool Has(uint8_t aType) const { return Locate(aType) != nullptr; }
+};
+
+/**
+ * This class provides methods to manipulate a Thread operational dataset.
+ * It maintains an internal buffer sized to accommodate the maximum possible dataset size.
+ */
+class OperationalDataset : public OperationalDatasetView
+{
+public:
+    OperationalDataset() : OperationalDatasetView(mBuffer) {}
+
+    // Delegate copy construction and assignment to the overloads taking an OperationalDatasetView.
+    OperationalDataset(const OperationalDataset & other) : OperationalDataset(static_cast<const OperationalDatasetView &>(other)) {}
+    OperationalDataset & operator=(const OperationalDataset & other)
+    {
+        return *this = static_cast<const OperationalDatasetView &>(other);
+    }
+
+    OperationalDataset(const OperationalDatasetView & view) : OperationalDatasetView(view) { CopyData(); }
+    OperationalDataset & operator=(const OperationalDatasetView & view)
+    {
+        static_cast<OperationalDatasetView &>(*this) = view;
+        CopyData();
+        return *this;
+    }
 
     /**
-     * Retrieves the delay timer from the dataset.
+     * Initializes the dataset by copying the provided data into an internal buffer.
      *
-     * @retval CHIP_NO_ERROR on success.
-     * @retval CHIP_ERROR_TLV_TAG_NOT_FOUND if no security policy is present in the dataset.
-     * @retval CHIP_ERROR_INVALID_TLV_ELEMENT if the TLV element is invalid.
+     * @param[in]   aData       The data to interpret as a Thread operational dataset.
+     *
+     * @retval CHIP_NO_ERROR                Successfully initialized the dataset.
+     * @retval CHIP_ERROR_INVALID_ARGUMENT  The dataset length @p aLength is too long or @p data is corrupted.
      */
-    CHIP_ERROR GetDelayTimer(uint32_t & aDelayMillis) const;
+    CHIP_ERROR Init(ByteSpan aData);
 
     /**
-     * This method sets the delay timer within the dataset.
+     * This method sets Thread active timestamp to the dataset.
```

Comment:
"to" is odd here.  "within" or "in" would be better?  Same for the other setters.  I know this is pre-existing; followup is fine.

Reply:
Did a pass over these descriptions, should be worded better now.